strawberryfields/backends/gaussianbackend/gaussiancircuit.py

```diff
@@ -249,7 +249,7 @@ def smeanxp(self):
         nmodes = self.nlen
         r = np.empty(2*nmodes)
         r[0:nmodes] = 2*self.mean.real
-        r[nmodes:2*nmodes] = 2*self.mean.real
+        r[nmodes:2*nmodes] = 2*self.mean.imag
```

Comment:
@nquesada was this a bug before?

Reply:
Looks like it was; since the convention is [x1,..., xN, p1,..., pN], the second half of the vector should contain the imaginary parts of alpha.

Highlights an issue with our existing backend tests. They are mostly state-based, which ensures that all backends agree with one another, but is missing unit tests which would catch bugs like this.

Reply:
Yup. It was a bug. It was detected by the tests introduced in this PR.